### PR TITLE
Add bytearray.starts_with function

### PR DESCRIPTION
**Pull request by Newt6611**
As its name, a  handy function to checks a ByteArray whether it start with a certain prefix

### CODE REVIEW

File: lib/aiken/primitive/bytearray.ak [hunk 1]
```diff
@@ -626,3 +626,28 @@ test to_hex_1() {
...
+
+test starts_with_2() {
+  let x = "Hello, World!"
+  starts_with(x, "World") == False
+}
+
+test starts_with_3() {
+  let x = ""
+  starts_with(x, "World") == False
+}
```
Comment (KtorZ) [inline lib/aiken/primitive/bytearray.ak:653]:
```suggestion
}

test starts_with_4() {
  starts_with("", "")
}

test starts_with_5() {
  starts_with("Hello, World!", "Hello, World!")
}

test starts_with_6() {
  !starts_with("Hello, World!", "Hello")
}
```

**Comment by KtorZ** [comment]
Huh, @Newt6611 ... I didn't mean to replace existing tests; I was simply adding a few extra ones to illustrate an issue 😅 ...

**Comment by Newt6611** [comment]
oh, ok haha, should I get those tests back?

**Comment by KtorZ** [comment]
Would be nice, they were relevant!

**Pull request merged**